pkg/database/connection.go

```diff
@@ -35,15 +35,15 @@ func establishConnection(driver, connectionString string) *sqlx.DB {
 		configs...)
 	if errRetry != nil {
 		log.Printf("Unable to connect to the database, error: %v", errRetry)
-		return nil
+		return nil, errRetry
 	}
 
 	err = otelsql.RegisterDBStatsMetrics(conn, otelsql.WithAttributes(semconv.DBSystemKey.String(driver)))
 	if err != nil {
 		log.Printf("Unable to instrument the DB properly, error: %v", err)
-		return nil
+		return nil, err
```

Comment:
Should the entire DB startup process fail if we cannot register metrics for the db connection?

Reply:
Hm... As I don't know well enough the implementation behind I would say we fail. If it results into problems, which it didn't yet, we can condition the failure to the metrics being enabled, so it fails when metrics are requested by the user.